src/components/Game.tsx

```diff
@@ -58,13 +58,14 @@ export const Game: FC = (): JSX.Element => {
 
   return (
     <>
-      {isGameOver && <h1>GAME OVER</h1>}
       <Header onStartNewGame={startNewGameHandler} onRevertStateBack={revertStateBackHandler} score={score} />
       <div className='board' style={{ width: BOARD_WIDTH, position: 'relative' }}>
         {isGameOver && <div className='game-over'>GAME OVER</div>}
-        {tiles.map(({ id: key, value, position }) => (
-          <Tile {...{ key, value, position }} />
-        ))}
+        {tiles
+          .sort((a, b) => a.id - b.id) // Required for CSS transitions
+          .map(({ id, value, position, isMerged, isNew }) => (
+            <Tile key={id} {...{ value, position }} isMerged={isMerged} isNew={isNew} />
```

Comment:
Assign all the properties in one style. We prefer assign each property separately